fhir-persistence-schema/docs/DB2SchemaMigration.md

```diff
@@ -0,0 +1,382 @@
+// Parameters are tied to the logical resource
+Table tbl = Table.builder(schemaName, tableName)
+    .setVersion(2)
+    .addTag(FhirSchemaTags.RESOURCE_TYPE, prefix)
+    .setTenantColumnName(MT_ID)
+    .addBigIntColumn(             ROW_ID,      false)
+    .addIntColumn(     PARAMETER_NAME_ID,      false)
+    .addVarcharColumn(         STR_VALUE, msb,  true)
+    .addVarcharColumn(   STR_VALUE_LCASE, msb,  true)
+    .addBigIntColumn(LOGICAL_RESOURCE_ID,      false)
+    .addIndex(IDX + tableName + "_PSR", PARAMETER_NAME_ID, STR_VALUE, LOGICAL_RESOURCE_ID)
+    .addIndex(IDX + tableName + "_PLR", PARAMETER_NAME_ID, STR_VALUE_LCASE, LOGICAL_RESOURCE_ID)
+    .addIndex(IDX + tableName + "_RPS", LOGICAL_RESOURCE_ID, PARAMETER_NAME_ID, STR_VALUE)
+    .addIndex(IDX + tableName + "_RPL", LOGICAL_RESOURCE_ID, PARAMETER_NAME_ID, STR_VALUE_LCASE)
+    .addPrimaryKey(PK + tableName, ROW_ID)
+    .setIdentityColumn(ROW_ID, Generated.BY_DEFAULT)
+    .addForeignKeyConstraint(FK + tableName + "_PNID", schemaName, PARAMETER_NAMES, PARAMETER_NAME_ID)
+    .addForeignKeyConstraint(FK + tableName + "_RID", schemaName, logicalResourcesTable, LOGICAL_RESOURCE_ID)
+    .setTablespace(fhirTablespace)
+    .addPrivileges(resourceTablePrivileges)
+    .enableAccessControl(this.sessionVariable)
+    .build(model);
+```
+
+If a new resource is added to the specification, the schema utility automatically provisions it on the next execution of the update schema actions. 
+
+Common reasons to modify the Resource's tables are: 
+- **Specification Change (Version-to-Version changes)**
+    - The Resources are saved as Blobs in the database, and are transparent to the FHIR Version changes. The changes from version-to-version are resilient to field add-remove changes, type changes, and resources additions.  This migration must be done manually.
+    - Resource removals must be done manually. 
+- **Column Attribute Changes (Space, Value Type)**
+    - As the column types, changes are changed in any column, a specific alter statement must be executed on the table, and constraints must be changed or relaxed. This migration must be done manually.
+- **Index - Add or Remove or Update**
+    - As indices are removed from the table definition, the removed indices must be manually dropped for each Resource table. 
+    - As indices are added to the table definition, the version of the table must be updated, and the index must be applied and updated manually.
+- **Constraint Updates** If there are Foreign Key updates, the changes must be applied manually and reflected in the code base. 
+- **Search Parameter Changes (Specification and Tenant)**
+    - If the the parameter type or code is changed, the PARAMETERS_NAME and the corresponding table must be updated to remove references to the removed parameter (based on code). 
+    - If there is a new SearchParameter with `code` newly added to the system, the Search Parameter value entry is only updated when the resource is updated. 
```

Comment:
```suggestion
    - If there is a new SearchParameter (with a new `code`) added to the system, the SearchParameter values are only updated if/when the resource is updated.
```

Reply:
We should probably have an open issue for this one...we've talked about ways to add a "re-index" feature but it hasn't become a top priority yet.

Reply:
FHIR Search - Search Re-indexing #789 is now open